torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -660,6 +671,48 @@ std::shared_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
   return std::make_shared<ProcessGroupNCCL::WorkNCCL>(devices);
 }
 
+struct checkFuture {
+  checkFuture(
+      std::shared_ptr<ProcessGroupNCCL::WorkNCCL> work,
+      std::vector<at::Tensor>& outputs,
+      int work_counter,
+      int input_size,
+      std::mutex& mutex)
```

Comment:
Avoid passing `work_counter` and `mutex` here. `work_counter` can be initialized to 0 in the constructor and can be a `std::atomic`.